chain/processor.go

```diff
@@ -138,21 +95,15 @@ func (b *StatelessBlock) Execute(
 
 			// Commit results to parent [TState]
 			tsv.Commit()
-
-			// Update key cache
-			if len(toCache) > 0 {
-				cacheLock.Lock()
-				for k := range toCache {
-					cache[k] = toCache[k]
-				}
-				cacheLock.Unlock()
-			}
 			return nil
 		})
 	}
 	if err := e.Wait(); err != nil {
 		return nil, nil, err
 	}
+	if err := f.HandleErrors(); err != nil {
```

Comment:
I would change this function to `Wait()`, it is more common for this sort of thing.